### DISCUSSION TITLE
Datagrid not visible in VisualStudio preview

### DISCUSSION THREAD

**Discussion by vmiwouter** [amwx/FluentAvalonia#266] [Q&A] [answered]
With the preview versions datagrid is not visible.

The designer gives as warning:
Unable to resolve type DataGrid from namespace https://github.com/avaloniaui Line 87, position .

Sample doesn't have this issue.

**Comment by amwx** [comment]
Have you referenced Avalonia.Controls.DataGrid in your project and made sure its the correct package version from upstream?

**Comment by vmiwouter** [comment]
I'm using the 11.0.0-preview4 via the nuget.

**Comment by amwx** [comment, marked as answer]
This is a weird one. Note that this only affects the previewer. This will not prevent the app from running and the DataGrid will still work.

 It seems what is associated with the xmlns `https://github.com/avaloniaui` is determined from styles or resources in `App.axaml` which is an issue with the previewer. Are you using VS or Rider? The DataGrid is essentially a separate entity and apparently needs a hard reference to its styles for it to get properly registered. (FATheme has a reference to the xaml file for DataGrid so not sure why it's not resolving).

As a workaround, add this to the Application.Resources in app.axaml (note that this has to be at the app level or it won't work). 
```Xaml
<ResourceDictionary>
	<ResourceDictionary.MergedDictionaries>
		<ResourceInclude Source="avares://FluentAvalonia/Styling/ControlThemes/BasicControls/DataGrid/DataGridStyles.axaml" />
	</ResourceDictionary.MergedDictionaries>
</ResourceDictionary>
```

Create an issue upstream (you can link to here). As I said above, this is something with the previewer so its out of my hands.

**Reply by vmiwouter** [reply]
I use VS, but this workaround works fine.
Thank you.
